src/main/kotlin/translations/identification/TranslationExternalAnnotationsArtifactsResolver.kt

```diff
@@ -0,0 +1,130 @@
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.demonwav.mcdev.translations.identification
+
+import com.demonwav.mcdev.util.invokeAndWait
+import com.demonwav.mcdev.util.invokeLater
+import com.intellij.codeInsight.ExternalAnnotationsArtifactsResolver
+import com.intellij.codeInsight.externalAnnotation.location.AnnotationsLocation
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.PathManager
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.AnnotationOrderRootType
+import com.intellij.openapi.roots.impl.libraries.LibraryEx
+import com.intellij.openapi.roots.libraries.Library
+import com.intellij.openapi.roots.libraries.ui.OrderRoot
+import com.intellij.openapi.roots.ui.configuration.libraryEditor.ExistingLibraryEditor
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.VirtualFileManager
+import org.jetbrains.concurrency.AsyncPromise
+import org.jetbrains.concurrency.Promise
+import org.jetbrains.concurrency.resolvedPromise
+
+class TranslationExternalAnnotationsArtifactsResolver : ExternalAnnotationsArtifactsResolver {
+    override fun resolve(project: Project, library: Library, mavenId: String?): Boolean {
+        if (!Util.isOurFakeMavenLocation(mavenId)) {
+            return false
+        }
+
+        return invokeAndWait {
+            doResolve(library)
+        }
+    }
+
+    override fun resolve(project: Project, library: Library, annotationsLocation: AnnotationsLocation): Boolean {
+        if (annotationsLocation != Util.fakeMavenLocation) {
+            return false
+        }
+        return invokeAndWait {
+            doResolve(library)
+        }
+    }
+
+    override fun resolveAsync(project: Project, library: Library, mavenId: String?): Promise<Library> {
+        if (!Util.isOurFakeMavenLocation(mavenId)) {
+            return resolvedPromise(library)
+        }
+
+        val promise = AsyncPromise<Library>()
+        ApplicationManager.getApplication().executeOnPooledThread {
+            invokeLater {
+                doResolve(library)
+                promise.setResult(library)
+            }
+        }
+
+        return promise
+    }
+
+    private fun doResolve(library: Library): Boolean {
+        if (library !is LibraryEx || library.isDisposed) {
+            return false
+        }
+
+        return runWriteAction {
+            val annotationsPath = findAnnotationsPath(false)
+                ?: findAnnotationsPath(true)
+                ?: return@runWriteAction false
+
+            val editor = ExistingLibraryEditor(library, null)
+            val type = AnnotationOrderRootType.getInstance()
+            editor.getUrls(type).forEach { editor.removeRoot(it, type) }
```

Comment:
Won't this remove existing external annotations ?
Could there be a way to only add those provided by the plugin ?

Reply:
I copied from the `ExternalAnnotationsRepositoryResolver` here, it seems to work fine without it though so I'll remove it.